src/traces/parcoords/parcoords.js

```diff
@@ -456,18 +456,20 @@ module.exports = function(root, svg, parcoordsLineLayers, styledData, layout, ca
 
     parcoordsLineLayer
         .each(function(d) {
-
             if(d.viewModel) {
-                if(d.lineLayer) d.lineLayer.update(d);
-                else d.lineLayer = lineLayerMaker(this, d);
+                if((!d.lineLayer) ||
+                    (callbacks)) { // recreate in case of having callbacks e.g. restyle. Should we test for callback to be a restyle?
+                    d.lineLayer = lineLayerMaker(this, d);
+                } else d.lineLayer.update(d);
 
-                d.viewModel[d.key] = d.lineLayer;
+                if(d.key) {
```

Comment:
when is `d.key` falsey? Makes sense not to set `d.viewModel[d.key]` if it doesn't have a `key` but what about `d.lineLayer.render`?

Reply:
Not sure if we always get a good d.key, so I thought we could check the index before setting the array.
In respect to your comment I put the render call outside the if statements. 
Now with the increase of delay, we've got success for the rest of the tests.